tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/listener/rag/LlmMailPrioritizationClassifierListener.java

```diff
@@ -169,16 +174,24 @@ private Mono<Void> processAddedEvent(MailboxEvents.Added addedEvent) {
         Username username = addedEvent.getUsername();
         MailboxSession session = mailboxManager.createSystemSession(username);
 
-        return getUserDisplayName(username)
-            .flatMap(userDisplayName -> Flux.from(messageIdManager.getMessagesReactive(addedEvent.getMessageIds(), FetchGroup.FULL_CONTENT, session))
-                .flatMap(messageResult -> buildUserPrompt(messageResult, username, userDisplayName)
-                    .flatMap(userPrompt -> Mono.from(metricFactory.decoratePublisherWithTimerMetric("llm-mail-prioritization-classifier", callLlm(systemPrompt, userPrompt)))
-                        .doOnError(e -> LOGGER.error("LLM call failed for messageId {} in mailboxId {} of user {}",
-                            messageResult.getMessageId().serialize(), addedEvent.getMailboxId().serialize(), addedEvent.getUsername().asString(), e)))
-                    .map(this::isActionRequired)
-                    .flatMap(actionRequired -> actionRequired ? addNeedsActionKeyword(messageResult, addedEvent.getMailboxId(), session) : Mono.empty()), ReactorUtils.LOW_CONCURRENCY)
-                .doFinally(signal -> mailboxManager.endProcessingRequest(session))
-                .then());
+        return aiNeedActionsSettingEnabled(username)
+            .flatMap(needsActionEnabled -> getUserDisplayName(username)
+                .flatMap(userDisplayName -> Flux.from(messageIdManager.getMessagesReactive(addedEvent.getMessageIds(), FetchGroup.FULL_CONTENT, session))
+                    .flatMap(messageResult -> buildUserPrompt(messageResult, username, userDisplayName)
+                        .flatMap(userPrompt -> Mono.from(metricFactory.decoratePublisherWithTimerMetric("llm-mail-prioritization-classifier", callLlm(systemPrompt, userPrompt)))
+                            .doOnError(e -> LOGGER.error("LLM call failed for messageId {} in mailboxId {} of user {}",
+                                messageResult.getMessageId().serialize(), addedEvent.getMailboxId().serialize(), addedEvent.getUsername().asString(), e)))
```

Comment:
7 (**7 !**) indent level spotted.

It more than deserve a refactoring in order to make this humanly readable...